### PR TITLE
Fixed undefined in procedures that didn't specify validation

### PR DESCRIPTION
**Pull request by Emulebest**
resolves #147

### CODE REVIEW

**Comment by shlomi-nx** [comment]
This is related to #134 but not exactly it.

I did not mean to create unvalidated procedures, rather, procedures which validates they get no input or return no output. Like a `void` type.

I am not entirely sure I want to keep `unvalidated` procedures around, I much rather them get validated to `any object` (https://swagger.io/docs/specification/data-models/data-types/#object look for `free-form` object)

**Comment by Emulebest** [comment]
> This is related to #134 but not exactly it.
> 
> I did not mean to create unvalidated procedures, rather, procedures which validates they get no input or return no output. Like a `void` type. The following three permutations are needed:
> 
> 1. `(void)->a`
> 2. `(a)->void`
> 3. `(void)->void`
> 4. `(a)->b` - what we currently support
> 
> I am not entirely sure I want to keep `unvalidated` procedures around, if someone needs this, I much rather the procedures get validated to `any object` (https://swagger.io/docs/specification/data-models/data-types/#object look for `free-form` object)

Regarding this one, I defaulted all the procedures that didn't specify a validation scheme to `AnyType` which is basically is an `any` type. Seems like I chose naming for this stuff incorrectly

**Comment by shlomi-nx** [comment]
I have seen what you have did, and that was my comment exactly. Any is assignable from anything, so this is a type which is always valid. That is not what I want, I want a type like `Nothing` that fails on everything.

File: papiea-engine/__tests__/api_docs.test.ts [hunk 1]
```diff
@@ -123,4 +129,47 @@ describe("API Docs Tests", () => {
...
+            await providerApi.post('/', provider);
+            const apiDoc = await apiDocsGenerator.getApiDocs();
+            console.dir(apiDoc.paths);
+            expect(apiDoc.paths[`/services/${ provider.prefix }/${ provider.version }/procedure/${ procedure_id }`]
+                .post
+                .requestBody
+                .content["application/json"]
+                .schema
+                .properties
+                .input['$ref']).toEqual(`#/components/schemas/AnyValue`);
```
Comment (shlomi-nx) [inline papiea-engine/__tests__/api_docs.test.ts]: This should be `NoValue` which represents the absence of any value.

File: papiea-engine/src/api_docs/api_docs_generator.ts [hunk 2]
```diff
@@ -291,8 +291,18 @@ export default class ApiDocsGenerator {
         };
     }
 
+    processEmptyValidation(proc_def: any, sig: Procedural_Signature) {
+        if (!sig.argument) {
+            proc_def.requestBody.content["application/json"].schema.properties.input.$ref = `#/components/schemas/AnyValue`
+        }
+        if (!sig.result) {
+            proc_def.responses["200"].content["application/json"].schema.properties.input.$ref = `#/components/schemas/AnyValue`
```
Comment (shlomi-nx) [inline papiea-engine/src/api_docs/api_docs_generator.ts]: Same here

**Pull request merged**